src/Fl_GIF_Image.cxx

```diff
@@ -588,9 +463,323 @@ void Fl_GIF_Image::load_gif_(Fl_Image_Reader &rdr)
+        Fl::warning("Fl_GIF_Image: %s invalid LZW-initial code size %d.\n", rdr.name(), CodeSize);
+      }
+      CodeSize++;
+
+      // Fix images w/o color table. The standard allows this and lets the
+      // decoder choose a default color table. The standard recommends the
+      // first two color table entries should be black and white.
+
+      if (ColorMapSize == 0) { // no global and no local color table
+        Fl::warning("%s does not have a color table, using default.\n", rdr.name());
+        BitsPerPixel = CodeSize - 1;
+        ColorMapSize = 1 << BitsPerPixel;
+        CMap.Red[0] = CMap.Green[0] = CMap.Blue[0] = 0;    // black
+        CMap.Red[1] = CMap.Green[1] = CMap.Blue[1] = 255;  // white
+        for (int i = 2; i < ColorMapSize; i++) {
+          CMap.Red[i] = CMap.Green[i] = CMap.Blue[i] = (uchar)(255 * i / (ColorMapSize - 1));
+        }
+      }
+
+      // Workaround for broken GIF files...
+      BitsPerPixel = CodeSize - 1;
+      if (1 << BitsPerPixel <= 256)
+        ColorMapSize = 1 << BitsPerPixel;
+
+      // Fix transparent pixel index outside ColorMap (Issue #271)
+      if (has_transparent && transparent_pixel >= ColorMapSize) {
+        for (int k = ColorMapSize; k <= transparent_pixel; k++)
+          CMap.Red[k] = CMap.Green[k] = CMap.Blue[k] = 0xff; // white (color is irrelevant)
+        ColorMapSize = transparent_pixel + 1;
+      }
+
+#if (0) // TEST/DEBUG: fill color table to maximum size
+      for (int i = ColorMapSize; i < 256; i++) {
+        CMap.Red[i] = CMap.Green[i] = CMap.Blue[i] = 0; // black
+      }
+#endif
+
+      CHECK_ERROR
+
+      // now read the LZW compressed image data
+
+      Image = new uchar[Width*Height];
+      lzw_decode(rdr, Image, Width, Height, CodeSize, ColorMapSize, Interlace);
+      if (ld()) return; // CHECK_ERROR aborted already
+
+      // Notify derived class on loaded image data
+
+      GIF_FRAME gf(frame, ScreenWidth, ScreenHeight, XPos, YPos, Width, Height, Image);
+      gf.disposal(dispose, user_input ? -delay - 1 : delay);
+      gf.colors(ColorMapSize, background_color_index, has_transparent ? transparent_pixel : -1);
+      GIF_FRAME::CPAL cpal[256] = { { 0 } };
+      if (HasLocalColorTable)
+        gf.cpal = LocalColorTable;
+      else if (HasGlobalColorTable)
+        gf.cpal = GlobalColorTable;
+      else {
+        for (i=0; i < ColorMapSize; i++) {
+          cpal[i].r = CMap.Red[i]; cpal[i].g = CMap.Green[i]; cpal[i].b = CMap.Blue[i];
+        }
+        gf.cpal = cpal;
+      }
+#if (0) // TEST/DEBUG: output palette values
+      printf("palette:\n");
+        for (i=0; i<ColorMapSize; i++) {
+          printf("%d: #%02X%02X%02X\n", i, gf.cpal[i].r, gf.cpal[i].g, gf.cpal[i].b);
+      }
+#endif
+      on_frame_data(gf);
+
+      // We are done reading the image, now convert to xpm (first image only)
+      if (!frame) {
+        w(anim ? ScreenWidth : Width);
```

Comment:
If the first animated frame has an offset, our image data will be the size of the frame, but the GIF size will be larger ("ScreenWidth", "ScreenHeight"), so reading the pixel data will cause a memory read violation. This happens when the gif is resized.

Reply:
I think the commit below fixes this. It repositions the frame in a bigger canvas.

Reply:
I remember that this worked initially.
It was broken by an adaption of the code because of a change in FLTK, that made `Fl_Image::copy()`  a `const` method:

This was my commit that broke this case:
https://github.com/fltk/fltk/pull/375/commits/a2ed2d7b3e584fbe4cf8a83377ee89ec1042f8e2